src/rajawali/Object3D.java

```diff
@@ -649,8 +649,8 @@ protected void cloneTo(Object3D clone, boolean copyMaterial) {
 	public Object3D clone(boolean copyMaterial, boolean cloneChildren) {
 		Object3D clone = new Object3D();
 		cloneTo(clone, copyMaterial);
-		clone.setRotation(getRotation());
-		clone.setScale(getScale());
+        clone.setRotation(getRotation().clone());
```

Comment:
hi there
i can assure you it is bugged
that was the reason i went in and fixed it
when i scaled a model, all clones of it scaled also

On Fri, Mar 6, 2015 at 7:39 PM, Ian Thomas <notifications@github.com> wrote:

> In src/rajawali/Object3D.java
> <https://github.com/MasDennis/Rajawali/pull/1353#discussion_r25966348>:
>
> > @@ -649,8 +649,8 @@ protected void cloneTo(Object3D clone, boolean copyMaterial) {
> >  	public Object3D clone(boolean copyMaterial, boolean cloneChildren) {
> >  		Object3D clone = new Object3D();
> >  		cloneTo(clone, copyMaterial);
> > -		clone.setRotation(getRotation());
> > -		clone.setScale(getScale());
> > +        clone.setRotation(getRotation().clone());
>
> This is redundant as the setRotation method calls setAll on the vector
> which copies the internal double values.
>
>
> https://github.com/MasDennis/Rajawali/blob/master/src/rajawali/ATransformable3D.java#L212
>
> Which then calls
>
>
> https://github.com/MasDennis/Rajawali/blob/master/src/rajawali/math/vector/Vector3.java#L167
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MasDennis/Rajawali/pull/1353/files#r25966348>.
>